src/main/java/org/candlepin/subscriptions/metering/service/prometheus/PrometheusMetricsProperties.java

```diff
@@ -35,50 +42,80 @@
 @ConfigurationProperties(prefix = "rhsm-subscriptions.metering.prometheus.metric")
 public class PrometheusMetricsProperties {
 
-  public static final String OPENSHIFT_PRODUCT_PROFILE_ID = "OpenShift";
+  @Autowired TagProfile tagProfile;
```

Comment:
I'm not a fan of field injection. Can we use constructor injection here?

Could use the constructor to store the result of `tagProfile.measurementsByTag(productTag).forEach(metric -> metrics.put(metric, openshift));` instead of recalculating on each call to `getSupportedMetricsForProduct`.

Reply:
Agreed on this. An oversight -- thanks!!!

DONE